flask/app/__init__.py

```diff
@@ -27,6 +28,9 @@ def create_app(config):
     app.config.from_object(config)
     app.json_encoder = DateTimeEncoder
 
+    for ex in default_exceptions:
+        app.register_error_handler(ex, handle_error)
```

Comment:
Can this be registered generically for all exceptions instead? I got an HTML response after disconnecting the database.

Reply:
See: 9d788b98472d4f96cb16de571a04991f7c9e238c 

Also added the handler as a blueprint. The drawback with the generic handler is similar to try catching an Exception, in this case all unhandled exceptions (non HTTP) will default to 500, but at least the error message is captured?

Reply:
That's what should happen! There's no other appropriate error code.